sdk/identity/identity/src/credentials/deviceCodeCredentialOptions.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { InteractiveCredentialOptions } from "./interactiveCredential";
+
+/**
+ * Provides the user code and verification URI where the code must be
+ * entered.  Also provides a message to display to the user which
+ * contains an instruction with these details.
+ */
+export interface DeviceCodeInfo {
+  /**
+   * The device code that the user must enter into the verification page.
+   */
+  userCode: string;
+
+  /**
+   * The verification URI to which the user must navigate to enter the device
+   * code.
+   */
+  verificationUri: string;
+
+  /**
+   * A message that may be shown to the user to instruct them on how to enter
+   * the device code in the page specified by the verification URI.
+   */
+  message: string;
+}
+
+/**
+ * Defines the signature of a callback which will be passed to
+ * DeviceCodeCredential for the purpose of displaying authentication
+ * details to the user.
+ */
+export type DeviceCodePromptCallback = (deviceCodeInfo: DeviceCodeInfo) => void;
+
+/**
+ * Defines options for the InteractiveBrowserCredential class for NodeJS.
+ */
+export interface DeviceCodeCredentialOptions extends InteractiveCredentialOptions {
+  /**
+   * Optional timeout after which this credential will stop waiting for the user to authenticate.
+   */
+  timeoutInMs?: number;
```

Comment:
Does this need to be configurable?

Reply:
Python offers a way to define the timeout: https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/identity/azure-identity/azure/identity/_credentials/device_code.py#L41 MSAL also offers this, of course.

I think I saw that a while ago, from which I made this issue: https://github.com/Azure/azure-sdk-for-js/issues/11670

I can remove this if it's not necessary, of course. Let me know what you think.

Reply:
On our call we agreed on removing the timeout.